src/walk-manifest.js

```diff
@@ -207,18 +216,21 @@ const walkPlaylist = function(options) {
     let resources = [];
     const manifest = {parent};
 
-    manifest.uri = uri;
-    manifest.file = path.join(basedir, path.basename(fsSanitize(uri)));
+    if (uri) {
+      manifest.uri = uri;
+      manifest.file = path.join(basedir, urlBasename(uri));
+    }
 
     // if we are not the master playlist
     if (dashPlaylist && parent) {
       manifest.file = parent.file;
       manifest.uri = parent.uri;
     } else if (parent) {
       manifest.file = path.join(
-        path.dirname(parent.file),
+        basedir,
+        path.dirname(path.relative(basedir, parent.file)),
```

Comment:
noticed that for deeper playlists we end up adding an extra output dir, ie `hls-fetcher` dir, as the dirname of the first playlist is the output dir.